app_module/src/main/java/com/enrique/apprater/AppRater.java

```diff
@@ -0,0 +1,139 @@
+
+    private int daysUntilPrompt = 3;
+
+    private int launchesUntilPrompt = 7;
+
+    private Context contexT;
+
+
+    public AppRater(Context context) {
+        contexT = context;
+        appPackageName = context.getBundleName();
+    }
+
+    /** Init method.
+     *
+     *
+     *
+     * @return Apprater object.
+     * @noinspection checkstyle:Indentation, checkstyle:Indentation
+     */
+    public AppRater init() {
+        DatabaseHelper databaseHelper = new DatabaseHelper(contexT);
+
+
+        Preferences prefs = databaseHelper.getPreferences(APP_RATER);
+        if (prefs.getBoolean(DONT_SHOW, false)) {
+            return null;
+        }
+
+        Preferences editor = prefs;
+        long launchcount = prefs.getLong(LAUNCH_COUNT, 0) + 1;
+        editor.putLong(LAUNCH_COUNT, launchcount);
+        Long datefirstLaunch = prefs.getLong(FIRST_LAUNCH, 0);
+        if (datefirstLaunch == 0) {
+            datefirstLaunch = System.currentTimeMillis();
+            editor.putLong(FIRST_LAUNCH, datefirstLaunch);
+        }
+        if ((launchcount >= launchesUntilPrompt)
+                && (System.currentTimeMillis() >= datefirstLaunch + (daysUntilPrompt * 24 * 60 * 60 * 1000))) {
+            showRateDialog(contexT, editor);
+
+        }
+        editor.flushSync();
+        return this;
+    }
+
+    /**  ShowrateDialog.
+     *
+     *
+     * @param mcontext
+     *
+     * @param editor Member variable of preferences
+     * @return   AppraterObject.
+     */
+
+    public AppRater showRateDialog(final Context mcontext, final Preferences editor) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(mcontext);
+        builder.setMessage(mcontext.getString(ResourceTable.String_dialog_text, apptitlE));
+        builder.setTitle(mcontext.getString(ResourceTable.String_rate) + " " + apptitlE);
+        builder.setNegativeButton(mcontext.getString(ResourceTable.String_no), new IDialog.ClickedListener() {
+            @Override
+            public void onClick(IDialog dialog, int id) {
+                dialog.destroy();
+            }
+        });
+
+        builder.setNeutralButton(mcontext.getString(ResourceTable.String_dontask), new IDialog.ClickedListener() {
+            @Override
+            public void onClick(IDialog dialog, int id) {
+                if (editor != null) {
+                    editor.putBoolean(DONT_SHOW, true);
+                    editor.flushSync();
+                }
+                dialog.destroy();
+            }
+        });
+
+        builder.setPositiveButton(mcontext.getString(ResourceTable.String_rate), new IDialog.ClickedListener() {
+            @Override
+            public void onClick(IDialog dialog, int id) {
+                Intent intent = new Intent();
+                Operation operation = new Intent.OperationBuilder()
+                        .withUri(Uri.parse("https://appgallery.cloud.huawei.com/appDetail?pkgName=" + appPackageName))
```

Comment:
Add this is hardcode string value into a string constant variable (private static final) and use the constant in the code.